### PR TITLE
Allow nested array access.

### PR DESCRIPTION
**Pull request by liverbool**
| Q               | A
| --------------- | -----
| Branch?         | 1.1
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | #7935 
| License         | MIT

Allow to access nested array, see https://symfony.com/doc/current/components/property_access.html for more.

### CODE REVIEW

File: src/Sylius/Bundle/ResourceBundle/Controller/ParametersParser.php [hunk 1]
```diff
@@ -89,9 +115,8 @@ private function parseRequestValue($parameter, Request $request)
      */
     private function parseRequestValueExpression(string $expression, Request $request)
     {
-        $expression = preg_replace_callback('/(\$\w+)/', function ($matches) use ($request) {
-            $variable = $request->get(substr($matches[1], 1));
-
+        $expression = preg_replace_callback('/(\$[\w\[\]]+)/', function ($matches) use ($request) {
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/Controller/ParametersParser.php]: Isn't it breaking support for simple arguments like `$foo`?

File: src/Sylius/Bundle/ResourceBundle/Controller/ParametersParser.php [hunk 2]
```diff
@@ -81,6 +82,31 @@ private function parseRequestValue($parameter, Request $request)
...
 
+    /**
+     * @param string $parameter
+     * @param Request $request
+     *
+     * @return mixed
+     */
+    private function getRequestValue($parameter, Request $request)
+    {
+        preg_match('/[a-z0-9]+/i', $parameter, $match);
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/Controller/ParametersParser.php]: `/^(?P<variable>[^\[]++)(?P<property>(?:\[[^\]\[]++\])++)?$/` regexp might do better here, you will get both of the things you need using one `preg_match` call.

Comment (pamil) [reply src/Sylius/Bundle/ResourceBundle/Controller/ParametersParser.php]: You can access them through `$match['variable']` and `$match['property']`.

**Comment by liverbool** [comment]
i'll be back.

**Comment by Zales0123** [comment]
@liverbool it the meantime, **SyliusResourceBundle** was extracted to the [separate repository](https://github.com/Sylius/SyliusResourceBundle/), if you still have an intention to implement this feature, you should do it there :) Thanks! 🖖

**Pull request closed without merging**